### ISSUE TITLE
no-extra-parens: one IIFE form is still required by ESLint to have parens

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#1004]
The following code:

``` js
(function () {
    // some code
})();
```

with setting `no-extra-parens` turned on returns an error:

```
error  Gratuitous parentheses around expression  no-extra-parens
```

This pattern should be allowed for the same reason the pattern:

``` js
(function () {
    // some code
}());
```

is allowed, especially that the first form is much more popular.
